lib/sorbet-rails/gem_plugins/friendly_id_plugin.rb

```diff
@@ -0,0 +1,20 @@
+# typed: strict
+class FriendlyIdPlugin < SorbetRails::ModelPlugins::Base
+  sig { implementation.params(root: Parlour::RbiGenerator::Namespace).void }
+  def generate(root)
+    return unless @model_class.singleton_class.included_modules.include?(::FriendlyId)
+
+    model_rbi = root.create_class(
+      model_class_name,
+      superclass: 'ApplicationRecord'
+    )
+
+    # Friendly method definition
+    # https://github.com/norman/friendly_id/blob/53f1ca01c5d9e71b62812028ff009fd138fd41b5/lib/friendly_id/base.rb#L216
+    model_rbi.create_method(
```

Comment:
you likely want to use a new method in `model_util` called `add_relation_query_method` to add this method to all kind of relations: active record relation, association relation, association collection proxy.